### ISSUE TITLE
[BaselineRGBDataset] Optimize __getitem__ performance

### ISSUE THREAD

**Issue by kapiblue** [JankowskiDaniel/Neural-Deep-Retina#84]
It seems to me that we can optimize __getitem__ performance by changing this:

```python
# Stack three consecutive grayscale images
        images = []
        for i in range(self.subseq_len):
            x = self.X[idx + i]
            x = torch.from_numpy(x)
            images.append(x)
        # Stack images along the channel dimension
        x = torch.stack(images, dim=0)  # Shape will be (3, H, W)
```
to this:
```python
x = self.curr_X[idx : idx + self.subseq_len]
```
and transforming `self.X` and `self.Y` to torch Tensors after reading and preprocessing (Remark: scikit-learn normalization doesn't work for tensors, just numpy arrays)

@JankowskiDaniel please confirm this change as the author of the original code. I hope I have not overlooked something. 

I can say that training seems to be around 40% faster using the second option :)

**Comment by JankowskiDaniel** [comment]
Good point, refactored in https://github.com/JankowskiDaniel/Neural-Deep-Retina/commit/eb66bcabd1c63d727793e6f85d69e60f20f69bd7
